### ISSUE TITLE
First run issues

### ISSUE THREAD

**Issue by boxed** [guillp/requests_oauth2client#5]
https://guillp.github.io/requests_oauth2client/#as-auth-handler is incorrect. The example code is:

```py
api_client = ApiClient(
    "https://myapi.local/resource",
    auth=OAuth2ClientCredentials(
        oauth2client, scope="myscope", resource="https://myapi.local"
    ),
)
```
but `OAuth2ClientCredentials` doesn't take any non-keyword argument. Is also must take `token_url, `client_id` and `client_secret` keyword arguments. I also don't understand what the `resource` argument is supposed to be.

If I try:

```py
    api_client = ApiClient(
        "https://foo/endpoint",
        auth=OAuth2ClientCredentials(
            token_url="https://foo/oauth-v1/token",
            scope=scope,
            resource="https://foo/endpoint",
            client_id=CLIENT_ID,
            client_secret=CLIENT_SECRET,
        ),
    )
```
    
I get a crash in requests in `prepare_auth` when it's trying to do:
```py
            # Allow auth to make its changes.
            r = auth(self)
```
because `auth` here is the `OAuth2ClientCredentials` instance and it's not callable.

**Comment by boxed** [comment]
Hm.. Seems the problem was that I imported the wrong `OAuth2ClientCredentials`. There was one from another library.

Maybe the docs should say `OAuth2ClientCredentialsAuth` and not `OAuth2ClientCredentials`? Because that class actually exists in this library, while `OAuth2ClientCredentials` does not.

**Comment by guillp** [comment]
Thanks for the report. You are correct, the doc should say `OAuth2ClientCredentialsAuth`. I'll fix it.
Feel free to report any other mistake or typo in the docs :)

**Comment by guillp** [comment]
fixed in #6
